### PR TITLE
Add docstrings and types for MambaCache

### PR DESCRIPTION
**Pull request by koayon**
# What does this PR do?

Adds some typing to the MambaCache and some docstrings for the IDE.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

## Who can review?

Just a small developer Quality of Life one, would love it if you could take a look @ArthurZucker and @younesbelkada!

### CODE REVIEW

File: src/transformers/models/mamba/modeling_mamba.py [hunk 1]
```diff
@@ -35,7 +35,6 @@
 from ...utils.import_utils import is_causal_conv1d_available, is_mamba_ssm_available
 from .configuration_mamba import MambaConfig
```
Comment (koayon) [inline src/transformers/models/mamba/modeling_mamba.py:37]:
```suggestion


```

File: src/transformers/models/mamba/modeling_mamba.py [hunk 2]
```diff
@@ -57,11 +57,29 @@
 _CONFIG_FOR_DOC = "MambaConfig"
 
 
-from ..deprecated._archive_maps import MAMBA_PRETRAINED_MODEL_ARCHIVE_LIST  # noqa: F401, E402
+from ..deprecated._archive_maps import (  # noqa: F401, E402
+    MAMBA_PRETRAINED_MODEL_ARCHIVE_LIST,
+)
```
Comment (koayon) [inline src/transformers/models/mamba/modeling_mamba.py:62]:
```suggestion
from ..deprecated._archive_maps import MAMBA_PRETRAINED_MODEL_ARCHIVE_LIST # noqa: F401, E402
```

File: src/transformers/models/mamba/modeling_mamba.py [hunk 3]
```diff
@@ -57,11 +57,27 @@
 _CONFIG_FOR_DOC = "MambaConfig"
 
 
-from ..deprecated._archive_maps import MAMBA_PRETRAINED_MODEL_ARCHIVE_LIST  # noqa: F401, E402
+from ..deprecated._archive_maps import MAMBA_PRETRAINED_MODEL_ARCHIVE_LIST # noqa: F401, E402
```
Comment (koayon) [inline src/transformers/models/mamba/modeling_mamba.py:60]:
```suggestion
from ..deprecated._archive_maps import MAMBA_PRETRAINED_MODEL_ARCHIVE_LIST  # noqa: F401, E402
```

**Comment by koayon** [comment]
A test is currently failing referencing the `generate_whisper` file which as far as I can tell has nothing to do with this PR? 

image

**Review by ArthurZucker — APPROVED** [review]
Sure! Thanks

**Comment by ArthurZucker** [comment]
Could you run `make fixup` for the remaining test?

**Comment by ArthurZucker** [comment]
feel free to also rebase on main

**Comment by koayon** [comment]
Yeah good idea, have rebased!

**Comment by koayon** [comment]
@ArthurZucker the tests are now passing and it's good to merge - thanks! 🙏

**Comment by ArthurZucker** [comment]
Please be a tad bit patient 🤗

**Pull request merged**